mesonbuild/mdist.py

```diff
@@ -29,14 +31,25 @@
 from .scripts.meson_exe import run_exe
 
 archive_choices = ['gztar', 'xztar', 'zip']
+
+def __archive_permutations() -> T.Tuple[str]:
+    permutations: T.List[str] = []
+    for i in range(len(archive_choices) + 1):
+        for subset in itertools.permutations(archive_choices, i):
+            if len(subset) > 0:
+                permutations.append(','.join(subset))
+
+    return tuple(permutations)
+
+archive_permutations = __archive_permutations()
 archive_extension = {'gztar': '.tar.gz',
                      'xztar': '.tar.xz',
                      'zip': '.zip'}
 
 def add_arguments(parser):
     parser.add_argument('-C', default='.', dest='wd',
                         help='directory to cd into before running')
-    parser.add_argument('--formats', default='xztar', choices=archive_choices,
+    parser.add_argument('--formats', default='xztar',
                         help='Comma separated list of archive types to create.')
```

Comment:
You can add list of supported archives in the help message.

Reply:
Meant to do that.